examples/plugins/altemplate/src/log.rs

```diff
@@ -0,0 +1,80 @@
+/* Copyright (C) 2018 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// same file as rust/src/applayertemplate/logger.rs except
+// different paths for use statements
+// open_object using altemplate instead of just template
+// Addition of JsonBuilderWrapper to look like a rust app-layer
+// even if we must use C API for JsonBuilder (because of its rust repr)
+
+use super::template::TemplateTransaction;
+use std::ffi::CString;
+use suricata::cast_pointer;
+use suricata::jsonbuilder::JsonError;
+use suricata_sys::jsonbuilder::{JsonBuilder, jb_set_string, jb_close, jb_open_object};
+
+use std;
+
+// syntax sugar around C API of JsonBuilder to feel like a normal app-layer in log_template
+pub struct JsonBuilderWrapper {
+    inner: *mut JsonBuilder,
+}
+
+impl JsonBuilderWrapper {
```

Comment:
New crate, `suricata_plugin_utils` :)

But this is good for now, at least the plugin is not hand writing its own bindings which is what I didn't want.

Reply:
I propose to create a new ticket after this gets merged for new crates core and suricata_plugin_utils